pkg/agent/controller/networkpolicy/priority_test.go

```diff
@@ -253,25 +254,88 @@ func TestReassignPriorities(t *testing.T) {
 	pa.Release(64354)
 	expectedOFMap = map[uint16]types.Priority{
-		64360: p111, 64358: p1122, 64357: p1131, 64355: p1142,
+		64360: p110, 64358: p1121, 64357: p1130, 64355: p1141,
 	}
 	expectedPriorityMap := map[types.Priority]uint16{
-		p111: 64360, p1122: 64358, p1131: 64357, p1142: 64355,
+		p110: 64360, p1121: 64358, p1130: 64357, p1141: 64355,
 	}
 	expectedSorted := []uint16{64355, 64357, 64358, 64360}
 	assert.Equalf(t, expectedOFMap, pa.ofPriorityMap, "Got ofPriorityMap %v, expected %v", pa.ofPriorityMap, expectedOFMap)
 	assert.Equalf(t, expectedPriorityMap, pa.priorityMap, "Got priorityMap %v, expected %v", pa.priorityMap, expectedPriorityMap)
 	assert.Equalf(t, expectedSorted, pa.sortedOFPriorities, "Got sortedOFPriorities %v, expected %v", pa.sortedOFPriorities, expectedSorted)
 }
+
+func TestRevertUpdates(t *testing.T) {
+	tests := []struct {
+		name                string
+		insertionPoint      uint16
+		extraPriority       types.Priority
+		originalPriorityMap map[types.Priority]uint16
+		originalOFMap       map[uint16]types.Priority
+		originalSorted      []uint16
+	}{
+		{
+			"single-update-up",
+			9999,
+			p1121,
+			map[types.Priority]uint16{p1120: 9999, p1130: 9998},
+			map[uint16]types.Priority{9999: p1120, 9998: p1130},
+			[]uint16{9998, 9999},
+		},
+		{
+			"multiple-updates-up",
+			9997,
+			p1131,
+			map[types.Priority]uint16{
+				p1120: 9999, p1121: 9998, p1130: 9997, p1140: 9996, p1141: 9995, p1160: 9994, p1161: 9993},
+			map[uint16]types.Priority{
+				9999: p1120, 9998: p1121, 9997: p1130, 9996: p1140, 9995: p1141, 9994: p1160, 9993: p1161},
+			[]uint16{9993, 9994, 9995, 9996, 9997, 9998, 9999},
+		},
+		{
+			"single-update-down",
+			9999,
+			p1121,
+			map[types.Priority]uint16{p1120: 10000, p1130: 9999},
+			map[uint16]types.Priority{10000: p1120, 9999: p1130},
+			[]uint16{9999, 10000},
+		},
+		{
+			"multiple-updates-down",
+			9998,
+			p1131,
+			map[types.Priority]uint16{
+				p1120: 10000, p1121: 9999, p1130: 9998, p1140: 9997, p1141: 9996, p1160: 9995},
+			map[uint16]types.Priority{
+				10000: p1120, 9999: p1121, 9998: p1130, 9997: p1140, 9996: p1141, 9995: p1160},
+			[]uint16{9995, 9996, 9997, 9998, 9999, 10000},
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			pa := newPriorityAssigner(func(p types.Priority) uint16 {
+				return tt.insertionPoint
+			})
+			for ofPriority, p := range tt.originalOFMap {
+				pa.updatePriorityAssignment(ofPriority, p)
+			}
+			_, _, revertFunc, _ := pa.GetOFPriority(tt.extraPriority)
+			revertFunc()
+			assert.Equalf(t, tt.originalPriorityMap, pa.priorityMap, "Got priorityMap %v, expected %v", pa.priorityMap, tt.originalPriorityMap)
+			assert.Equalf(t, tt.originalOFMap, pa.ofPriorityMap, "Got ofPriorityMap %v, expected %v", pa.ofPriorityMap, tt.originalOFMap)
+			assert.Equalf(t, tt.originalSorted, pa.sortedOFPriorities, "Got sortedOFPriorities %v, expected %v", pa.sortedOFPriorities, tt.originalSorted)
```

Comment:
The messages here are not really useful. You just end up printing duplicate information as `assert` already prints the expected and actual maps.

I suggests just using something like "Unexpected priorityMap", "Unexpected ofPriorityMap", ...

Reply:
Done